activerecord/lib/active_record/migration.rb

```diff
@@ -1375,6 +1397,24 @@ def parse_migration_filename(filename)
         File.basename(filename).scan(Migration::MigrationFilenameRegexp).first
       end
 
+      def validate_timestamp?
+        ActiveRecord.timestamped_migrations && ActiveRecord.validate_migration_timestamps
+      end
+
+      def valid_migration_timestamp?(version)
+        match_data = version.match(MIGRATION_TIMESTAMP_REGEX)
+        return false unless match_data
+
+        year, month, day, hour, minute, second = match_data.captures
+        begin
+          Time.new(year, month, day, hour, minute, second)
```

Comment:
Note that this is insufficient to detect things like `20200231...` (February 31st), as Ruby normalizes those edge cases by doing the math to calculate what the correct `Time` would be. The approach I used in the cop was to [check if the timestamp round trips.](https://github.com/rubocop/rubocop-rails/pull/1044/files#diff-be0de4bbd3c3e836cab455658b9873aea491bfe31f60ecf02e35a78dadd962efR51). We might not care though. 🤷‍♂️

Reply:
Makes sense to me to try to cover reasonable edge cases (this seems like one), so I'll add a check for that 👍  I didn't even think to check how you solved this already in the cop, thanks for the nudge 😂

Reply:
Thanks @sambostock ,  updated `valid_migration_timestamp?` to compare against the original version and added another test! I also changed it to use `Time.utc` over `Time.new` since that's technically more precise.